libslide/src/partial_evaluator.rs

```diff
@@ -56,12 +56,56 @@ mod tests {
     }
 
     partial_evaluator_tests! {
+        add:                            "1 + 2"     => "3"
+        add_nested_left:                "1 + 2 + a" => "a + 3"
+        // TODO: This doesn't work yet... we need to be smarter, maybe by rewritting associativity
+        // or flipping constants between passes.
+        // add_nested_right:               "a + 1 + 2" => "a + 3"
+        // add_nested_with_reorder:        "1 + a + 2" => "a + 3"
+
+        sub:                            "1 - 2"     => "-1"
+        sub_nested_left:                "1 - 2 - a" => "-1 - a"
+
+        mult:                           "2 * 3"     => "6"
+        mult_nested_left:               "2 * 3 * a" => "6 * a"
+
+        div:                            "6 / 2"     => "3"
+        div_nested_left:                "6 / 2 / a" => "3 / a"
+        div_associated:                 "6 / 2 / 3" => "1"
+
+        modulo:                         "6 % 4"     => "2"
+        modulo_nested_left:             "6 % 4 % a" => "2 % a"
+        modulo_associated:              "9 % 5 % 5" => "4" // (9 % 5) % 5
+
+        exp:                            "2 ^ 3"     => "8"
+        exp_nested_left:                "2 ^ 3 ^ a" => "2 ^ 3 ^ a"
+        exp_associated:                 "2 ^ 3 ^ 2" => "512"
+
+        posate:                         "+1"           => "1"
+        // TODO: unwrap parantheses further
+        posate_nested_1:                "+(b + c)" => "(b + c)"
+        posate_nested:                  "a + +(b + c)" => "a + (b + c)"
+
```

Comment:
add negative versions of these. The positive version is relatively imo and I still don't think it was necessary to add a unary operator for it. I think it is bad to allow our users to have a functiion  ++++5 without throwing an error saying they are an idiot.

Reply:
Can you elaborate by what you mean by negative versions? In particular, what rules would you like? I can think of `"-(_a - _b) -> _b - _a"`, anything else?

Reply:
-(b + c) => -b -c is all I was thinking. This can be added in another pr if you want

Reply:
I'll add it